tests/rt_cascading.erl

```diff
@@ -1084,6 +1102,50 @@ wait_exit(Pids, Timeout) ->
         end
     end, Mons).
 
+make_clusters(UnNormalClusterConfs) ->
+    ClusterConfs = lists:map(fun
+        ({Name, Size}) ->
+            {Name, Size, []};
+        ({_Name, _Size, _ConnectsTo} = E) ->
+            E
+    end, UnNormalClusterConfs),
+    DeployList = lists:foldl(fun
+        ({_Name, Size, _ConnectTo}, ConfAcc) ->
+            Conf = conf(),
+            AddToAcc = lists:duplicate(Size, {current, Conf}),
+            ConfAcc ++ AddToAcc
+    end, [], ClusterConfs),
+    Nodes = rt:deploy_nodes(DeployList),
+    lager:info("nodes deployed: ~p", [Nodes]),
+    {NamesAndNodes, []} = lists:foldl(fun
+        ({Name, Size, _ConnectTo}, {Clusters, NodesLeft}) ->
+            {ForCluster, NewNodesLeft} = lists:split(Size, NodesLeft),
+            {Clusters ++ [{Name, ForCluster}], NewNodesLeft}
+    end, {[], Nodes}, ClusterConfs),
+    NamesAndNodes = lists:map(fun({Name, ForClusterNodes}) ->
+        {Name, make_cluster(Name, ForClusterNodes)}
+    end, NamesAndNodes),
+    ok = lists:foreach(fun({Name, _Size, ConnectsTo}) ->
+        lists:foreach(fun(ConnectToName) ->
+            connect_rt(get_node(Name, NamesAndNodes), get_port(ConnectToName, NamesAndNodes), ConnectToName)
+        end, ConnectsTo)
+    end, ClusterConfs),
+    NamesAndNodes.
+
+make_cluster(Name, Nodes) ->
+    repl_util:make_cluster(Nodes),
+    _ = [repl_util:wait_until_is_leader(N) || N <- Nodes],
+    [ANode | _] = Nodes,
+    repl_util:name_cluster(ANode, Name),
+    Nodes.
+
+get_node(Name, NamesAndNodes) ->
+    [Node | _] = proplists:get_value(Name, NamesAndNodes),
+    Node.
+
+get_port(Name, NamesAndNodes) ->
+    get_cluster_mgr_port(get_node(Name, NamesAndNodes)).
+
 conf() ->
     [{lager, [
         {handlers, [
```

Comment:
At line 1206, `WaitTime` ought to be called `SleepTime` and carried through.

At line 1220, where we start checking return values from `exists/3` we should use a new function name like `wait_until_obj_exists/6`.